app/Http/Controllers/MessageController.php

```diff
@@ -45,33 +58,73 @@ public function store(Request $request)
         $message->player()->associate($this->session->player);
         $group->messages()->save($message);
 
-        // Return Created response
         return new CreatedResponse($message);
     }
 
     /**
-     * Get the messages from the given chat group.
+     * Show a message.
+     *
+     * @param MessageGroup $group
+     * @param Message $message
+     * @return Response|UnauthorizedResponse
+     */
+    public function show(MessageGroup $group, Message $message)
+    {
+        if (!$group->message_group_members->pluck('player_id')->contains($this->session->player_id)) {
+            return new UnauthorizedResponse();
+        }
+
+        return new Response($message);
+    }
+
+    /**
+     * Edit a message.
+     * TODO: Determine under which circumstances we should allow this (if at all).
+     *  Right now we allow all players to edit their own messages.
      *
+     * @param MessageGroup $group
+     * @param Message $message
      * @param Request $request
-     * @return Response
-     * @throws ValidationException
+     * @return UnauthorizedResponse|UpdatedResponse
      */
-    public function index(Request $request)
+    public function update(MessageGroup $group, Message $message, Request $request)
     {
-        \Validator::make($request->all(), [
-            'group_id' => 'required|int',
-        ])->validate();
+        $request->validate([
+            'message' => 'required|string',
+        ]);
 
-        if (!$group = MessageGroup::whereId($request->input('group_id'))->first()) {
-            throw ValidationException::withMessages(['Message group does not exist']);
+        if (
+            !$group->message_group_members->pluck('player_id')->contains($this->session->player_id) ||
+            $message->player != $this->session->player
+        ) {
+            return new UnauthorizedResponse();
         }
 
-        if (!$group->message_group_members->pluck('player_id')->contains($this->session->player_id)) {
-            throw ValidationException::withMessages(['Player is not in the message group']);
-        }
+        $message->message = $request->get('message');
+        $message->save();
+
+        return new UpdatedResponse($message);
+    }
 
-        $messages = Message::whereMessageGroupId($request->input('group_id'))->get();
+    /**
+     * Delete a message.
+     * TODO: Determine under which circumstances we should allow this (if at all).
+     *  Right now we allow all players to delete their own messages.
+     * @param MessageGroup $group
```

Comment:
I don't think we should allow deleting messages at all

Reply:
Hmmmm, it could be interesting... it would definitely make (the act of) subterfuge more viable, and might make it harder to share things using screenshots. I think we should consider it (for the standard game), and meanwhile we could probably allow this as an option in game creation.